sql: TestTelemetry failed

sql.TestTelemetry [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4150708&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4150708&tab=artifacts#/) on master @ [912964e02ddd951c77d4f71981ae18b3894e9084](https://github.com/cockroachdb/cockroach/commits/912964e02ddd951c77d4f71981ae18b3894e9084):


```
=== RUN   TestTelemetry
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry1418443230
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTelemetry
    telemetry_test.go:45: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry1418443230
--- FAIL: TestTelemetry (74.29s)
=== RUN   TestTelemetry/planning/server
    telemetry.go:90: 
        testdata/telemetry/planning:178: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
        expected:
        sql.plan.lateral-join
        
        found:
        sql.plan.lateral-join
        sql.plan.subquery
        --- FAIL: TestTelemetry/planning/server (1.97s)
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (3.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/testdata/telemetry/planning
+++ b/pkg/sql/testdata/telemetry/planning
@@ -167,7 +167,6 @@ sql.plan.cte.recursive
 
 feature-allowlist
 sql.plan.lateral-join
-sql.plan.subquery.*
 ----
 
 feature-usage
@@ -179,6 +178,10 @@ feature-usage
 SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
 ----
 sql.plan.lateral-join
+
+feature-allowlist
+sql.plan.subquery.*
+----
 
 feature-usage
 SELECT 1 = (SELECT a FROM x LIMIT 1)

